GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/Utilities/VFSProcessRunner.m

```diff
@@ -0,0 +1,81 @@
+#import "VFSProcessRunner.h"
+
+@interface VFSProcessRunner()
+
+@property (strong) ProcessFactory processFactory;
+
+@end
+
+@implementation VFSProcessRunner
+
+- (instancetype)initWithProcessFactory:(ProcessFactory)processFactory
+{
+    if (processFactory == nil)
+    {
+        self = nil;
+    }
+    else if (self = [super init])
+    {
+        _processFactory = processFactory;
+    }
+    
+    return self;
+}
+
+/**
+ Runs an executable specified by path and args. If the executable could be run
+ successfully - output will contain the executable's combined stderr/stdout
+ output and the method returns YES. In case of failure, it returns NO and error
+ will hold the executable's combined stderr/stdout output.
+
+ @param path - specify full path to the executable.
+ @param args - specify any command line args to pass to the executable.
+ @param output - contains executable's output, if it was successfully run.
+ @param error - contains executable's output, if it exited with an error.
+ @return YES if the executable was successfully run, NO otherwise.
+ */
+- (BOOL)tryRunExecutable:(NSURL *)path
```

Comment:
Would it be simpler if output always contain stdout and error always contain stderr? Or maybe just have a single output parameter which contains the combined output?

Having two out params that contain the same data (and only differ based on success or not) seems more complicated.

Reply:
I will clean this up in an upcoming PR. Have made a note of this in the [TODO list](https://github.com/microsoft/VFSForGit/issues/1262).